### PR TITLE
Handle multiple apps

### LINKED ISSUE #3: Support getting all testers instead of specifying an app_id
Would like to go through all our testers and/or go through an array of app ids. To do a proper house cleaning of testers.

I'd be happy to try and submit a PR.

https://github.com/KrauseFx/fastlane-plugin-clean_testflight_testers/blob/5ac11aafa4426ec3666c241b649c5fcce4fcede7/lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb#L19
We are only removing for a specific app_identifier.

### PR DESCRIPTION
**Pull request by jetersen**
* Fetch all apps if no identifier specified.
* Handle multiple identifiers

fixes #3

### CODE REVIEW

File: lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb [hunk 1]
```diff
@@ -9,33 +9,51 @@ def self.run(params)
...
-        app_id = spaceship_app.apple_id
+        apps = []
+        unless app_identifiers.any?
+          UI.message("Fetching all apps, this might take a few minutes, depending on the number of apps 🤖")
+          apps = Spaceship::Application.all
+        else
+          UI.message("Fetching the #{app_identifiers.size} apps, you specified, give me a moment 🤖")
+          app_identifiers.each do |app_identifier|
+            app = Spaceship::Application.find(app_identifier)
+            unless app
```
Comment (jetersen) [inline lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb:25]: Could this be more Ruby like or is it good enough?

File: lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb [hunk 2]
```diff
@@ -45,12 +63,12 @@ def self.run(params)
         end
       end
 
-      def self.remove_tester(tester, app_id, dry_run)
+      def self.remove_tester(tester, app, dry_run)
```
Comment (jetersen) [inline lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb]: I hardly remember ids, so lets pass app object and show app name when printing and use apple_id when deleting.

File: lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb [hunk 3]
```diff
@@ -9,33 +9,51 @@ def self.run(params)
...
+              UI.message("Couldn't find app '#{app_identifier}' on the account of '#{params[:username]}' on iTunes Connect 😵")
+            else
+              apps << app
+            end
+          end
+          UI.user_error!("💥 Failed to find all apps specified 💥") unless app_identifiers.size == apps.size
+        end
 
-        all_testers = Spaceship::TestFlight::Tester.all(app_id: app_id)
         counter = 0
```
Comment (jetersen) [inline lib/fastlane/plugin/clean_testflight_testers/actions/clean_testflight_testers_action.rb:34]: We went from mere 4 testers being removed on our most popular test app
To now 164 when rolling over all apps 👍 
Only took 27 seconds to go through 🤖 . Yet for a human.... 😱

**Review by jetersen — COMMENTED** [review]
Feels like we need a 'Are you sure' before actually removing testers? 🤔 

image

**Comment by jetersen** [comment]
@KrauseFx when you have time 👍 It's been pretty solid for house cleaning.

**Comment by jetersen** [comment]
@KrauseFx will the PR be open for merging after I fix conflict or should I not even bother 👎 We definitely been using this feature for our use case.

**Comment by KrauseFx** [comment]
Hey @casz, sorry for the late reply, thanks for your PR 👍  Quick question: as a workaround, can users just call the plugin multiple times for each app, right?

**Comment by jetersen** [comment]
Yes, they could, but why not give them the power 👍 

Here we gave them 3 options.
specify one and we clean the single.
specify multiple in comma separated and we clean the multiple.
specify none and retrieve all and we clean ALL 👍 (Can't remember half of our bundle ids, not without looking)

**Comment by jetersen** [comment]
@KrauseFx I'll try and rebase and hope you will pick this up! Hate to see this go to waste

**Comment by jetersen** [comment]
@KrauseFx rebased
To answer your question again, yes you can just pass one at a time but not all of can remember every thing apple id that we maintain 👍

**Comment by jetersen** [comment]
@KrauseFx travis is very different from circle CI now it seems 😢

**Pull request closed without merging**